frontend/src/App.js

```diff
@@ -17,44 +19,53 @@ import PlayPage from "main/pages/PlayPage";
 import NotFoundPage from "main/pages/NotFoundPage";
 
 function App() {
-
   const { data: currentUser } = useCurrentUser();
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Don't derive another state from currentUser. This will cause out of sync state where currentUser data is logged in but isLoading is still true. Just read from the data directly during render.